OSDOCS-5605: Added a module to detail AWS LZ cluster limitations

dfitzmau opened the pull request:
Change Management for 4.12 updates


[OSDOCS-5605](https://issues.redhat.com/browse/OSDOCS-5605)

Version(s):
4.12 and 4.13

Link to docs preview:
[Cluster limitations in AWS Local Zones](https://57736--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-localzone.html#cluster-limitations-local-zone_installing-aws-localzone)

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

mtulio reviewed (changes requested):
This document aims to provide information on what will be changed from a regular installation or be informed about the Local Zones limitations **before installing**, so users can be prepared and aware of it.

dfitzmau commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,26 @@
+* Change the the cluster-wide network MTU to a lower value than the cluster's edge worker nodes. For example, set the cluster MTU to `1300` bytes and set the OVN-Kubernetes plugin to `1200` bytes. This configuration ensures that edge worker nodes in one Local Zone region can communicate with nodes in another Local Zone region.
+* Enable the Application Load Balancer and disable Network Load Balancer, Classic Local Balancer, and Nat Gateways for your cluster. This configuration causes the Ingress Controller to route user-work traffic to an AWS Load Balancer (ALB).
+**  If you want to use a public subnet located in the outpost instance for the ALB, you need to remove the special tag, `kubernetes.io/cluster/.*-outposts: owned`, that the installation program set during VPC creation. By applying this configuration, you do not need to create new services of type `LoadBalancer`, Network Load Balancer, for your cluster.
+* Set `gp2-csi` as the default storage class for your cluster by issuing the following command in your terminal:
```
Removed these lines.

mtulio reviewed (commented):
Could you please review all the statements based on the card's goal?

1) From the card:

> ...telling the customers the existing AWS Limitations and changes from the default when using that feature.

When reading the document I understand that the user[1] must change, but also the installer will change[2]. But it's not the case. The goal is to **inform**  the user about the limitations **provided by** AWS when using the Local Zones features. It's not something the user might need to act, or installer will change.

The statements seems not to be concise with Local Zones. The "installation program..." will not "Sets the `gp3-csi` storage class for your cluster" or "Enables the Network Load Balancer (NLB)".

Could you please review those statements with the provided on https://issues.redhat.com/browse/OSDOCS-5605 ? Let me know if I need more clarification on those topics for the card.

2 ) s/Outposts/Local Zones/

dfitzmau commented:
> Could you please review all the statements based on the card's goal?
> 
> 1. From the card:
> 
> > ...telling the customers the existing AWS Limitations and changes from the default when using that feature.
> 
> When reading the document I understand that the user[1] must change, but also the installer will change[2]. But it's not the case. The goal is to **inform** the user about the limitations **provided by** AWS when using the Local Zones features. It's not something the user might need to act, or installer will change.
> 
> The statements seems not to be concise with Local Zones feature. The "installation program..." will not "Sets the `gp3-csi` storage class for your cluster" or "Enables the Network Load Balancer (NLB)".
> 
> Could you please review those statements with the provided on https://issues.redhat.com/browse/OSDOCS-5605 ? Let me know if I need more clarification on those topics for the card.
> 
> 2. s/Outposts/Local Zones/

Hi @mtulio . I rehashed the content. It's a bit more generalized. Please let me know your thoughts.

mtulio commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="cluster-limitations-local-zone_{context}"]
+= Cluster limitations in AWS Local Zones
+
+Some limitations exist when you attempt to deploy a cluster with a default installation configuration in Amazon Web Services (AWS) Local Zones. If you do not not configure your cluster for deployment in AWS Local Zones, edge worker nodes might not process user workloads. 
```
```suggestion
Some limitations exist when you attempt to deploy a cluster with a default installation configuration in Amazon Web Services (AWS) Local Zones.
```

dfitzmau replied:
Updated.

mtulio commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,14 @@
+
+The following list details limitations when deploying a cluster in AWS Local Zones:
+
+- A higher cluster-wide network maximum transmission unit (MTU) value than the MTU specified for edge worker nodes can impact how edge worker nodes in your cluster communicate. Settings a lower cluster-wide network MTU can resolve this issue. 
```
```suggestion
- The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is 1300, thus the cluster-wide network MTU is changed according to the network plugin used on the deployment.
```

dfitzmau replied:
For style guide requirements, I had to slightly tweak the text.

What do you think of the following change?

> The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in a region is `1300`. This causes the cluster-wide network MTU to change according to the network plugin that is used on the deployment.

mtulio replied:
OK.
fore reference to "the Region":
https://docs.aws.amazon.com/local-zones/latest/ug/how-local-zones-work.html
```text
Generally, the Maximum Transmission Unit (MTU) is as follows:
(...)
- 1300 bytes between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region.
```

dfitzmau replied:
Thanks, updated.

mtulio commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,14 @@
+
+- A higher cluster-wide network maximum transmission unit (MTU) value than the MTU specified for edge worker nodes can impact how edge worker nodes in your cluster communicate. Settings a lower cluster-wide network MTU can resolve this issue. 
+- The Network Load Balancer (NLB), the Classic Load Balancer, and Nat Gateways can impact the deployment of your cluster in an AWS Local Zone. AWS Outposts do not support these networking components. Deploying a cluster in an AWS Local Zones environment requires the Application Load Balancer.
+- The `gp3-csi` storage class is not globally available in Local Zones. AWS Outposts do not support Amazon Block Store (EBS) `gp3` volumes. AWS Outposts support the `gp2-csi` storage class.
```
```suggestion
- The AWS Elastic Block Storage (EBS) `gp3`, the default for node volumes and Storage Class on AWS deployments, is not globally available on Local Zones locations. The nodes running in Local Zones are deployed with `gp2` EBS volume by default, and the StorageClass `gp2-csi` must be set when creating workloads on Local Zone nodes.
```

dfitzmau replied:
Updated to :+1: 
> The `gp3` AWS Elastic Block Storage (EBS), which is the default storage class for node volumes and AWS deployments, is not globally available on Local Zone locations. By default, the nodes running in Local Zones are deployed with the `gp2` EBS volume. The `gp2-csi` storage class must be set when creating workloads on Local Zone nodes.

mtulio replied:
>  which is the default storage class for node volumes and AWS deployments, 

This statement is not clear/correct. The storage class is a cluster component, and the node volumes are not related to Storage Class, it shares the same technology (EBS Volume) but for Kube world is not related. Having said that, the order I suggested makes difference:
- "the default for node volumes 
- and Storage Class on AWS deployments,"

dfitzmau replied:
Thanks, updated.

mtulio commented:
Two other considerations about the section:
1) it would be better to place right before or after the prerequisites; (currently it's after the opt-in that is an action to start the deployment)
2) On "Outposts" we added it as an "Important" box. It seems to be a good idea to have on LZ too as it's something to be considered before using the feature.

dfitzmau commented:
> Two other considerations about the section:
> 
> 1. it would be better to place right before or after the prerequisites; (currently it's after the opt-in that is an action to start the deployment)
> 2. On "Outposts" we added it as an "Important" box. It seems to be a good idea to have on LZ too as it's something to be considered before using the feature.

Thanks for the review.

I added the section after the prereq section. I also surrounded the text with an IMPORTANT notice. Please let me know your thoughts.

mtulio commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,14 @@
+The following list details limitations when deploying a cluster in AWS Local Zones:
+
+- A higher cluster-wide network maximum transmission unit (MTU) value than the MTU specified for edge worker nodes can impact how edge worker nodes in your cluster communicate. Settings a lower cluster-wide network MTU can resolve this issue. 
+- The Network Load Balancer (NLB), the Classic Load Balancer, and Nat Gateways can impact the deployment of your cluster in an AWS Local Zone. AWS Outposts do not support these networking components. Deploying a cluster in an AWS Local Zones environment requires the Application Load Balancer.
```
> Thanks for the review.
> 
> I added the section after the prereq section. I also surrounded the text with an IMPORTANT notice. Please let me know your thoughts.

It's perfect, thanks.

dfitzmau replied:
I'll leave out this sentence. Not sure what you mean by we are also mentioning it at the end of the page as a day-2 guide, right ? I've provided a link to ALB on https://github.com/openshift/openshift-docs/pull/57063/files

mtulio commented:
> Thanks for the review.
> 
> I added the section after the prereq section. I also surrounded the text with an IMPORTANT notice. Please let me know your thoughts.

It's perfect, thanks

mtulio commented:
Just reviewed it again. It's good, just small adjustments so we will be able to move on.

makentenza commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,17 @@
+
+- The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in a region is `1300`. This causes the cluster-wide network MTU to change according to the network plugin that is used on the deployment.
+- Network resources such as Network Load Balancer (NLB), Classic Load Balancer, and Nat Gateways are not supported in AWS Local Zones.
+- The `gp3` AWS Elastic Block Storage (EBS), which is the default storage class for node volumes and AWS deployments, is not globally available on Local Zone locations. By default, the nodes running in Local Zones are deployed with the `gp2` EBS volume. The `gp2-csi` storage class must be set when creating workloads on Local Zone nodes.
```
@mtulio can we include a link to the docs where we explain how to create a Storage Class on day-2?

mtulio replied:
@makentenza for sure. Pointing to our documentation we have this section: https://docs.openshift.com/container-platform/4.12/storage/understanding-persistent-storage.html#pvc-storage-class_understanding-persistent-storage

I strongly believe we can provide one example/day-2 docs using Local Zones and gp2-csi StorageClass, as our existing documentation is generic for AWS[1]. We are doing similar work exemplifying the deployment in Local Zones[2], it can be appended or based on it. WDYT?

[1] https://docs.openshift.com/container-platform/4.12/storage/generic-ephemeral-vols.html
[2] https://github.com/openshift/openshift-docs/pull/57063

dfitzmau replied:
I can move this update to https://github.com/openshift/openshift-docs/pull/57063

dfitzmau replied:
Add the link to the Storage Class doc

makentenza reviewed (commented):
Just a quick note from my side

yunjiang29 commented:
LGTM, cc @Phaow for storage part review

Phaow commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,17 @@
+
+- The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is `1300`. This causes the cluster-wide network MTU to change according to the network plugin that is used on the deployment.
+- Network resources such as Network Load Balancer (NLB), Classic Load Balancer, and Nat Gateways are not supported in AWS Local Zones.
+- The AWS Elastic Block Storage (EBS) `gp3` is the default for node volumes and for the storage class on AWS deployments. This storage block is not globally available on Local Zone locations. By default, the nodes running in Local Zones are deployed with the `gp2` EBS volume. The `gp2-csi` storage class must be set when creating workloads on Local Zone nodes.
```
```suggestion
- The AWS Elastic Block Storage (EBS) `gp3` type volume is the default for node volumes and for the default storage class set on AWS Openshift clusters. This volume type is not globally available on Local Zone locations. By default, the nodes running in Local Zones are deployed with the `gp2` EBS volume. The `gp2-csi` storageclass must be set when creating workloads on Local Zone nodes.
```
Will be better.

dfitzmau replied:
Thanks, @Phaow . Please let me know your thoughts.

Phaow replied:
@dfitzmau Thanks for the update!

Phaow commented:
LGTM.

dfitzmau commented:
/label peer-review-needed

dfitzmau commented:
/remove-label peer-review-needed

dfitzmau commented:
/label peer-review-needed

JoeAldinger commented:
/remove-label peer-review-needed
/label peer-review-in-progress

JoeAldinger reviewed (commented):
Suggestion that might help with clarity. 
/remove-label peer-review-in-progress
/label peer-review-done

dfitzmau commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,17 @@
+
+- The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is `1300`. This causes the cluster-wide network MTU to change according to the network plugin that is used on the deployment.
+- Network resources such as Network Load Balancer (NLB), Classic Load Balancer, and Nat Gateways are not supported in AWS Local Zones.
+- The AWS Elastic Block Storage (EBS) `gp3` type volume is the default for node volumes and the default for the storage class set on AWS OpenShift clusters. This volume type is not globally available on Local Zone locations. By default, the nodes running in Local Zones are deployed with the `gp2` EBS volume. The `gp2-csi` `StorageClass` must be set when creating workloads on Local Zone nodes.
```
Updated but went with `For an {product-title} cluster on AWS` because OpenShift on its own or without the preceding "Red Hat" tag might cause stylistic issues.

dfitzmau commented:
/remove-hold

dfitzmau commented:
/label merge-review-needed

sheriff-rh commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,17 @@
+// Module included in the following assemblies:
+//
+// * installing/installing-aws-localzone.adoc
+
```
Don't forget to add a `:content-type:` - this should be a CONCEPT or REFERENCE, up to you.

dfitzmau replied:
I'm not sure how I missed this. Thank you! Updated.

sheriff-rh reviewed (commented):
One small request before merge, otherwise looks good!

dfitzmau commented:
/label merge-review-needed

jboxman-rh commented on modules/cluster-limitations-local-zone.adoc:
```diff
@@ -0,0 +1,19 @@
+The following list details limitations when deploying a cluster in AWS Local Zones:
+
+- The Maximum Transmission Unit (MTU) between an Amazon EC2 instance in a Local Zone and an Amazon EC2 instance in the Region is `1300`. This causes the cluster-wide network MTU to change according to the network plugin that is used on the deployment.
+- Network resources such as Network Load Balancer (NLB), Classic Load Balancer, and Nat Gateways are not supported in AWS Local Zones.
```
@dfitzmau, afaik this is _NAT_, not _Nat_. (Network Address Translation)

dfitzmau replied:
Thanks, @jboxman-rh . I updated this text based on your feedback. 

You're correct: https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-gateway.html

jboxman-rh commented:
/cherry-pick enterprise-4.13

jboxman-rh commented:
/cherry-pick enterprise-4.12
